app/models/user.rb

```diff
@@ -1,8 +1,10 @@
 class User < ActiveRecord::Base
   has_secure_password
   validates_confirmation_of :password
+  validates :password, length: { minimum: 8 }
   validates :password, :name, :email, presence: true
   validates :email, uniqueness: true
+  validates_format_of :email, with: /.+@.+\..+/i
```

Comment:
Prefer the new style validations `validates :column, format: value` over `validates_format_of`.